addons/mail/static/src/new/discuss/channel_member_list.js

```diff
@@ -0,0 +1,46 @@
+/** @odoo-module **/
+
+import { Component } from "@odoo/owl";
+import { useMessaging } from "@mail/new/messaging_hook";
+import { PartnerImStatus } from "./partner_im_status";
+
+export class ChannelMemberList extends Component {
+    setup() {
+        this.messaging = useMessaging();
+        this.members = this.getMembersByStatus();
+        this.onlineMembers = this.members.onlines.sort((p1, p2) => (p1.name < p2.name ? -1 : 1));
+        this.offlineMembers = this.members.offlines.sort((p1, p2) => (p1.name < p2.name ? -1 : 1));
+    }
+
+    getMembersByStatus() {
+        const orderedOnlineMembers = [];
+        const orderedOfflineMembers = [];
+        for (const member of this.props.thread.channelMembers) {
+            const memberImStatus = this.messaging.state.partners[member.partnerId].im_status;
+            member.im_status = memberImStatus;
+            if (member.partnerId !== this.messaging.state.user.partnerId) {
+                member.hasOpenChatFromAvatarClick = true;
+            } else {
+                member.hasOpenChatFromAvatarClick = false;
+            }
+            if (memberImStatus === "online") {
+                orderedOnlineMembers.push(member);
+            } else {
+                orderedOfflineMembers.push(member);
+            }
+        }
+        return { onlines: orderedOnlineMembers, offlines: orderedOfflineMembers };
+    }
```

Comment:
It would be much better to have a getter in `thread_model.js`.
So that it's just `thread.onlineMembers` and `thread.offlineMembers`.